#! /usr/bin/env python

PACKAGE='ev3'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("UpdateRate", int_t, 1,  "Update rate",	 15, 1,	50)

gen.add("Vel", int_t, 1,  "Velocity",	 600, 1,	1540)
gen.add("Accel", int_t, 1,  "Acceleration",	 1500, 1,	99999)
gen.add("Margin", int_t, 1,  "Position margin",	 5, 1,	99999)
gen.add("Scale", double_t, 1,  "Scale",	 -1, -99999,	99999)
gen.add("PosKp", double_t,  1, "Position Kp", 2.0,  0,  99999)
gen.add("PosKi", double_t,  1,	"Position Ki", 0,	  0,  99999)
gen.add("PosKd", double_t,  1,	"Position Kd", 0,	  0,  99999)
gen.add("VelKp", int_t,  1, "Linear Velocity Kp", 800,	  0,  80000)
gen.add("VelKi", int_t,  1,	"Linear Velocity Ki", 30,	    0,  80000)
gen.add("VelKd", int_t,  1,	"Linear Velocity Kd", 5,	    0,  80000)
gen.add("AngleFactor", double_t,  1,	"Angle conversion factor", -0.0404,	    -99999,  99999)
gen.add("AngleOffset", double_t,  1,	"Angle conversion offset", -82.181,	    -99999,  99999)

exit(gen.generate(PACKAGE, "servo_reconfig_node", "servoParams"))
